src/maker/server2.rs

```diff
@@ -383,6 +360,47 @@ fn handle_client_taproot(maker: &Arc<Maker>, stream: &mut TcpStream) -> Result<(
             }
         };
 
+        let swap_id = match &message {
+            TakerToMakerMessage::GetOffer(msg) => Some(msg.id.clone()),
+            TakerToMakerMessage::SwapDetails(msg) => Some(msg.id.clone()),
+            TakerToMakerMessage::SendersContract(msg) => Some(msg.id.clone()),
+            TakerToMakerMessage::PrivateKeyHandover(msg) => Some(msg.id.clone()),
+        }
+        .ok_or_else(|| MakerError::General("Message missing swap_id"))?;
+
+        // Get or create connection state for this swap id
+        let mut connection_state = {
+            let mut ongoing_swaps = maker.ongoing_swap_state.lock()?;
+
+            match &message {
+                TakerToMakerMessage::GetOffer(_) => {
+                    // TODO: Right now sync_offerbook is sending GetOffer with an empty string swap_id
+                    // Refactor to not create a connection state when swap id is an empty string
+                    // Just return the Offer
+                    let (state, _) = ongoing_swaps.entry(swap_id.clone()).or_insert_with(|| {
+                        log::info!(
+                            "[{}] Creating new connection state for {:?}",
+                            maker.config.network_port,
+                            &swap_id
+                        );
+                        (ConnectionState::default(), Instant::now())
+                    });
+                    state.clone()
```

Comment:
I think this is a bug. If we create a connection state at every get offer, then it will be easily DOS'd. And we are not clearing the state until privkey handover, so if a taker just asks for an offer, but never does a swap, this state will remain forever. In fact, this will create a state every time takers do market discovery.

Instead, the connection state should only be created when a swap is commenced, i.e, from `SendersContract`. And clear out at the end of `PrivateKeyHandover` processing.

Corollary: No connection state should ever exist without a non-empty `swap_id`.

Reply:
Created the issues for tracking